packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -682,10 +699,20 @@ class _PageViewState extends State<PageView> {
         if (notification.depth == 0 && widget.onPageChanged != null && notification is ScrollUpdateNotification) {
           final PageMetrics metrics = notification.metrics;
           final int currentPage = metrics.page.round();
-          if (currentPage != _lastReportedPage) {
-            _lastReportedPage = currentPage;
+          if (currentPage != _lastOnPageChangedIndex) {
+            _lastOnPageChangedIndex = currentPage;
             widget.onPageChanged(currentPage);
           }
+        } else if (notification.depth == 0 && widget.pageDidChange != null && notification is ScrollEndNotification) {
+          if (!widget.pageSnapping) {
```

Comment:
Could this apply to pageSnapping equal to false? if not, it will be more clear to mentions this in the doc of pageDidChange